File: C/strong_number.c
#include <stdio.h>

int factorial(int d) {
    int fact = 1;
    for (int i = 1; i <= d; i++) fact *= i;
    return fact;
}

int main() {
    int n;
    printf("Enter a number: ");
    if (scanf("%d", &n) != 1) return 0;

    if (n < 0) {
        printf("Not a strong number\n");
        return 0;
    }

    int temp = n;
    int sum = 0;

    while (temp > 0) {
        int d = temp % 10;
        sum += factorial(d);
        temp /= 10;
    }

    if (sum == n) printf("Strong number\n");
    else printf("Not a strong number\n");

    return 0;
}

File: Python/strong_number.py
def main():
    n = int(input("Enter a number: "))
    temp = n

    total = 0
    if n < 0:
        print("Not a strong number")
        return

    while temp > 0:
        d = temp % 10
        fact = 1
        for i in range(1, d + 1):
            fact *= i
        total += fact
        temp //= 10

    if total == n:
        print("Strong number")
    else:
        print("Not a strong number")


if __name__ == "__main__":
    main()

